lib/auth/clt.go

```diff
@@ -96,22 +96,19 @@ func NewClient(cfg client.Config, params ...roundtrip.ClientParam) (*Client, err
 		return nil, trace.Wrap(err)
 	}
 
-	// This logic is necessary for the client to force client to always send
-	// a certificate regardless of the server setting. Otherwise the client may pick
-	// not to send the client certificate by looking at certificate request.
-	if len(cfg.TLS.Certificates) != 0 {
```

Comment:
Is this no longer needed?

Reply:
This got moved to the `configure` function in `credentials.go` and is still called.